docs/source/references/configurations.rst

```diff
@@ -0,0 +1,57 @@
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+| database_connection_string          | LineaPy db connection string  | string  | `sqlite:///$LINEAPY_HOME_DIR/db.sqlite`    | `LINEAPY_DATABASE_CONNECTION_STRING`            |
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+| customized_annotation_folder        | user annotations folder       | Path    | `$LINEAPY_HOME_DIR/customized_annotations` | `LINEAPY_CUSTOMIZED_ANNOTATION_FOLDER`          |
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+| do_not_track                        | disable user analytics        | boolean | false                                      | `LINEAPY_DO_NOT_TRACK`                          |
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+| logging_level                       | logging level                 | string  | INFO                                       | `LINEAPY_LOGGING_LEVEL`                         |
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+| logging_file                        | logging file path             | Path    | `$LINEAPY_HOME_DIR/lineapy.log`            | `LINEAPY_LOGGING_FILE`                          | 
++-------------------------------------+-------------------------------+---------+--------------------------------------------+-------------------------------------------------+
+
+All LineaPy configuration items follow following naming convention; in configuration file, all variable name should be lower case with underscore, 
+all environmental variable name should be upper case with underscore and all CLI options should be lower case 
+For instance, following three will achieve the same thing to change the LineaPy base folder to :code:`/lineapy` and start ipython.
+
+- Adding :code:`{"home_dir": "/lineapy"}` in configuration file: and run :code:`lineapy ipython`
+- In environmental variable: :code:`export LINEAPY_HOME_DIR=/lineapy && lineapy ipython` 
+- In CLI options: :code:`lineapy --home-dir='/lineapy' ipython`
+
+LineaPy also provides a CLI command to generate the configuration file (as a json file) based on your environmental variables and CLI options for example:
+
+.. code:: bash  
+    
+    $ lineapy --home-dir='/lineapy' init 
```

Comment:
You should also include in the docs an example config it generates (this is also helpful for my reviewing! I was just checking to make sure that it's also yaml).